includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -148,14 +148,17 @@ public function get_items() {
 	 * @return array|\WP_Error The response, or WP_Error on failure.
 	 */
 	private function prepare_item_for_response( WP_Post $item ) {
-		$terms = get_the_terms( $item->ID, 'job_listing_type' );
+		$terms = [];
+		if ( get_option( 'job_manager_enable_types' ) ) {
+			// Only query for terms if the listing types are enabled.
+			$terms = get_the_terms( $item->ID, 'job_listing_type' );
```

Comment:
@fjorgemota, sorry for the late comment here, but after approving, I noticed something in the code. I think we probably should use the `wpjm_get_the_job_types` here because there's a filter there.